src/parser.rs

```diff
@@ -1421,62 +1410,101 @@ pub enum AttributeValue<'input> {
 /// An attribute in a `DebuggingInformationEntry`, consisting of a name and
 /// associated value.
 #[derive(Copy, Clone, Debug, Eq, PartialEq)]
-pub struct Attribute<'input> {
+pub struct Attribute<'input, Endian>
+    where Endian: Endianity
+{
     name: constants::DwAt,
-    value: AttributeValue<'input>,
+    value: AttributeValue<'input, Endian>,
 }
 
-impl<'input> Attribute<'input> {
+impl<'input, Endian> Attribute<'input, Endian>
+    where Endian: Endianity
+{
     /// Get this attribute's name.
     pub fn name(&self) -> constants::DwAt {
         self.name
     }
 
-    /// Get this attribute's value.
-    pub fn value(&self) -> AttributeValue<'input> {
+    /// Get this attribute's raw value.
+    pub fn value(&self) -> AttributeValue<'input, Endian> {
         self.value
     }
+
+    /// Get this attribute's normalized value.
+    ///
+    /// Attribute values can potentially be encoded in multiple equivalent forms,
+    /// and may have special meaning depending on the attribute name.  This method
+    /// converts the attribute value to a normalized form based on the attribute
+    /// name.
+    pub fn normalize(&self) -> AttributeValue<'input, Endian> {
```

Comment:
Perhaps getting nitpicky, but: `normalize` makes me think of an effectful operation to normalize the contained value so that whenever the value is accessed in the future, it is the normalized version. I think `normalized` would be more intuitive for what is actually happening here.

However, I think the default we present should be the cleaned up / pretty / normalized value. We should be handling DWARF's incidental complexity instead of forcing it on our users. In that light, how about renaming what is currently `value` to `raw_value` and what is currently `normalize` to `value`?